09-maze_solver/progress/internal/solver/animation.go

```diff
@@ -0,0 +1,44 @@
+package solver
+
+import (
+	"image"
+	"image/color/palette"
+
+	"golang.org/x/image/draw"
+)
+
+const (
+	// pathRatio is the proportion of path pixels in the total image, which consists of path + wall + treasure + entrance.
+	pathRatio = 0.4
+	// totalExpectedFrames is the number of frames we want in the output gif.
```

Comment:
// We won't get exactly 30, because pathRatio is approximate. But we'll get around that.